### PR TITLE
Launcher factory + flatten new UI initial screen

### PR DESCRIPTION
**Pull request by DanVanAtta**
- extract local launcher code so that updated UI can call the same thing. To do this we move the operative code to its own module, then we make the input an interface to decouple from the existing UI code. 

- New UI is updated to call this code and local game launches. Still partial as player selection rows are not parsed yet so we get a NPE pretty quickly. 

- New UI is updated to have a flatter initial screen, 'more options' button removed and the 'settings' and 'launch map creator' are now directly on the new ui initial screen. Now looks like this:

new_initial

### CODE REVIEW

File: src/main/java/games/strategy/engine/framework/startup/launcher/LauncherFactory.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        .collect(Collectors.toMap(PlayerCountrySelection::getPlayerName, PlayerCountrySelection::getPlayerType));
+
+    final Map<String, Boolean> playersEnabled = playerRows.stream()
+        .collect(Collectors.toMap(PlayerCountrySelection::getPlayerName, PlayerCountrySelection::isPlayerEnabled));
+
+    // we don't need the playerToNode list, the disable-able players, or the alliances
+    // list, for a local game
+    final PlayerListing pl =
+        new PlayerListing(null, playersEnabled, playerTypes, gameSelectorModel.getGameData().getGameVersion(),
+            gameSelectorModel.getGameName(), gameSelectorModel.getGameRound(), null, null);
```
Comment (ssoloff) [inline src/main/java/games/strategy/engine/framework/startup/launcher/LauncherFactory.java]: I know this is copied code, but I'm smelling the Builder pattern here to hide the `null` details and the long constructor parameter list. :smile:  Just something to keep in mind as this feature is fleshed out.

**Comment by ssoloff** [comment]
@DanVanAtta In light of #2877, I'll let you decide if you want to squash or rebase this PR.  My initial inclination was to squash it based on the last two commits, but I'm hedging on a rebase based on what I perceive to be your intention in the first three commits.  Vinnie Barbarino is so confused... :smile:

**Comment by DanVanAtta** [comment]
Thanks for review - I am mixing two scopes by extracting code and changing UI. Squash makes this clear. Rebase is not that great as the last two comments are "javadocs", which tells us very little and have negative value as commits living as individual commits in the history.

File: src/main/java/games/strategy/engine/framework/startup/launcher/LauncherFactory.java (hunk 1 above)
Comment (DanVanAtta) [reply src/main/java/games/strategy/engine/framework/startup/launcher/LauncherFactory.java]: Generally, agree. Handcrafted builders is a lot of overhead, medium/long term i would not mind seeing immutables lib to be incorporated to help generated builders with annotations.

**Pull request merged**